File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//创建服务
app.service("brandService",function ($http) {

    //搜索/分页
    this.search=function (page,size,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);

    }

    //删除
    this.dele=function (deleteid) {
        return $http.get('../brand/delete.do?ids='+deleteid)
    }

    //修改回显
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }

    //修改
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }

    //添加
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    }

    //下拉框数据
    this.selectOptionList=function () {
        return $http.post('../brand/selectOptionList.do');
    }

});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/content/service/BrandService.java
package com.pinyougou.content.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> findAll();

    PageResult getPageResult(int page , int size);

    void add(TbBrand tbBrand);

    TbBrand findOne(long id);

    void update(TbBrand tbBrand);

    void delete(long[] id);

    PageResult getPageResult(TbBrand brand, int page , int size);

    /**
     * 下拉框数据回显
     */

    public List<Map> selectOptionList();

}
